src/ol/webgl/ShaderBuilder.js

```diff
@@ -28,10 +28,13 @@ uniform mediump int u_hitDetection;
 const float PI = 3.141592653589793238;
 const float TWO_PI = 2.0 * PI;
 
+vec4 premultiplyAlpha(vec4 color) {
+  return vec4(color.rgb * color.a, color.a);
+}
+
 // this used to produce an alpha-premultiplied color from a texture
 vec4 samplePremultiplied(sampler2D sampler, vec2 texCoord) {
-  vec4 color = texture2D(sampler, texCoord);
-  return vec4(color.rgb * color.a, color.a);
+  return premultiplyAlpha(texture2D(sampler, texCoord));
```

Comment:
I'm wondering if we can just get rid of this `samplePremultiplied` function altogether now